### ISSUE TITLE
[BUG] Can't build on LINUX

### ISSUE THREAD

**Issue by snsvrno** [amethyst/amethyst#2162]
## Description

Building a new project using 0.13 from crates on Linux fails.

## Reproduction Steps

Attempting to build an empty project with the following TOML

```toml
[package]
name = "pong-rust"
version = "0.1.0"
authors = ["snsvrno <snsvrno@tuta.io>"]
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
amethyst = { version = "0.13", features = [ "vulkan" ] }
```
on linux using

```
cargo 1.41.0 (626f0f40e 2019-12-03)
UBUNTU 18.04.4 LTS
linux kernel 5.3.0-40-generic
```

## What You Expected to Happen

Project to build without errors.

## What Actually Happened

Build failed.

```
pong-rust · master± ⟩ cargo build
   Compiling err-derive v0.1.6
   Compiling font-kit v0.4.0
   Compiling gfx-hal v0.2.1
   Compiling ahash v0.2.18
error[E0433]: failed to resolve: could not find `__rt` in `quote`
   --> /home/snsvrno/.cargo/registry/src/github.com-1ecc6299db9ec823/err-derive-0.1.6/src/lib.rs:145:63
    |
145 | fn display_body(s: &synstructure::Structure) -> Option<quote::__rt::TokenStream> {
    |                                                               ^^^^ could not find `__rt` in `quote`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `err-derive`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by snsvrno** [comment]
I just noticed I was using an older version for some reason, updated to **0.14** and getting the same error.

**Comment by azriel91** [comment]
Needs the same fix as https://github.com/rust-lang-nursery/failure/pull/343 in `err-derive`, and a version bump in anything that uses it.

**Comment by Feijo** [comment]
I'm having the same problem on Windows 10, is there any workaround?

Adding `quote = "=1.0.2"` in `Cargo.toml` didn't do the trick for me.

> λ cargo run
    Updating crates.io index
error: failed to select a version for the requirement `quote = "= 1.0.2"`
  candidate versions found which didn't match: 1.0.3, 1.0.1, 1.0.0, ...
  location searched: crates.io index

**Comment by azriel91** [comment]
1.0.2 got yanked, so there's no more workaround.

[`err-derive` got fixed](https://gitlab.com/torkleyy/err-derive/issues/15). We need to bump the version in code for everyone to pick it up automatically, but you can run `cargo update` to pick up the fix now.

**Comment by omrerdn** [comment]
Let me drop a link, if workaround still needed : https://stackoverflow.com/questions/60576277/rust-amethyst-error-e0433-failed-to-resolve-could-not-find-rt-in-qu

**Comment by sigod** [comment]
It would be nice if this issue gets fixed.

**Comment by valkum** [comment]
Can we backport the PR to older releases?

**Comment by MGlolenstine** [comment]
If targeting `master` branch on the GitHub instead of "0.14.0", we get a different error.

```
   Compiling pong v0.1.0 (/home/notyours/rustProjects/pong)
error[E0432]: unresolved import `amethyst::ecs::prelude::Resources`
 --> src/main.rs:3:32
  |
3 |     ecs::prelude::{ReadExpect, Resources, SystemData},
  |                                ^^^^^^^^^
  |                                |
  |                                no `Resources` in `ecs::prelude`
  |                                help: a similar name exists in the module: `Resource`

warning: unused imports: `ReadExpect`, `SystemData`
 --> src/main.rs:3:20
  |
3 |     ecs::prelude::{ReadExpect, Resources, SystemData},
  |                    ^^^^^^^^^^             ^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0599]: no method named `with_clear` found for type `std::result::Result<amethyst_rendy::plugins::window::RenderToWindow, amethyst_config::ConfigError>` in the current scope
  --> src/main.rs:32:26
   |
32 |                         .with_clear([0.34, 0.36, 0.52, 1.0]),
   |                          ^^^^^^^^^^ method not found in `std::result::Result<amethyst_rendy::plugins::window::RenderToWindow, amethyst_config::ConfigError>`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0432, E0599.
For more information about an error, try `rustc --explain E0432`.
error: could not compile `pong`.

To learn more, run the command again with --verbose.
```

I'm on Arch Linux with latest updates

**Comment by azriel91** [comment]
@MGlolenstine `RenderToWindow::from_config_path` returns a `Result<_, _>`, adding `?` after the call will fix it.

**Comment by MGlolenstine** [comment]
@azriel91 Nice catch, thanks!

**Comment by lberrymage** [comment]
Would a 0.14.1 release be a bad method to roll something like this out to those not running on `master`?

**Comment by downfall85** [comment]
As omrerdn said, you can use the following workaround :
- Edit Cargo.lock
- Replace quote package 1.0.3 with the 1.0.2 version (the checksum is important)

```
[[package]]
name = "quote"
version = "1.0.2"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "053a8c8bcc71fcce321828dc897a98ab9760bef03a4fc36693c231e5b3216cfe"
dependencies = [
 "proc-macro2 1.0.9",
]

```
- Replace all "quote 1.0.3" with "quote 1.0.2" to update dependencies

**Comment by MGlolenstine** [comment]
@downfall85 Hasn't 1.0.2 been yanked?

**Comment by themeshpotato** [comment]
> 
> 
> As omrerdn said, you can use the following workaround :
> 
>     * Edit Cargo.lock
> 
>     * Replace quote package 1.0.3 with the 1.0.2 version (the checksum is important)
> 
> 
> ```
> [[package]]
> name = "quote"
> version = "1.0.2"
> source = "registry+https://github.com/rust-lang/crates.io-index"
> checksum = "053a8c8bcc71fcce321828dc897a98ab9760bef03a4fc36693c231e5b3216cfe"
> dependencies = [
>  "proc-macro2 1.0.9",
> ]
> ```
> 
>     * Replace all "quote 1.0.3" with "quote 1.0.2" to update dependencies

Thanks! Seems like I didn't have to replace the "quote 1.0.3" with "quote 1.0.2" after updating the version and checksum of the package. The cargo.lock-file was automatically updated.

**Comment by azriel91** [comment]
Heya, this should be fixed by using `amethyst 0.15.0` which has just been released.
